app/v2/controllers/requestParsers/RetrieveUkPropertyBusinessAnnualSummaryRequestParser.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers.requestParsers
+
+import javax.inject.Inject
+import v2.controllers.requestParsers.validators.RetrieveUkPropertyBusinessAnnualSummaryValidator
+import v2.models.domain.Nino
+import v2.models.request.retrieveUkPropertyBusinessAnnualSummary.{RetrieveUkPropertyBusinessAnnualSummaryRawData,
+  RetrieveUkPropertyBusinessAnnualSummaryRequest}
+
+class RetrieveUkPropertyBusinessAnnualSummaryRequestParser @Inject()(val validator: RetrieveUkPropertyBusinessAnnualSummaryValidator)
```

Comment:
Please rename new classes and files (especially if they're in packages with classes from other endpoints) so that they unambiguously identify what they are as there are many very similarly named endpoints in the API. 

Specifically:
- Include "Historic" to highlight that not for one of the existing non-historic endpoints
- Include "Fhl" (or NonFhl) as there will be non-FHL endpoints too.
- AnnualSubmission (not AnnualSummary) just to follow the convention in this API
- No need for "Business"

So in this case something like 
`RetrieveHistoricFhlUkPropertyAnnualSubmissionRequestParser`